components/snap_recovery/src/services.rs

```diff
@@ -301,10 +370,21 @@ impl<ER: RaftEngine> RecoverData for RecoveryService<ER> {
                 Err(e) => error!("failed to get store id"; "error" => ?e),
             };
 
-            let _ = sink.success(resp).await;
+            resp
         };
 
-        self.threads.spawn_ok(task);
+        self.abort_last_recover_region(format!("recover_region by {}", ctx.peer()));
+        let (state, task) = RecoverRegionState::wrap_task(task);
+        *self.last_recovery_region_rpc.lock().unwrap() = Some(state);
```

Comment:
maybe move this line into `self.abort_last_recover_region`?
it seems still race here in some corner cases.